skfda/preprocessing/dim_reduction/_fpca.py

```diff
@@ -99,11 +101,16 @@ def __init__(
         weights: Optional[Union[ArrayLike, WeightsCallable]] = None,
         components_basis: Optional[Basis] = None,
     ) -> None:
-        self.n_components = n_components
-        self.centering = centering
-        self.regularization = regularization
-        self.weights = weights
-        self.components_basis = components_basis
+        self._n_components = n_components
```

Comment:
@vnmabus, I changed the names of the "private" fields so that they start with an underscore since the linter counted them as public and complained about the number of public fields.
However, it might be a good idea to make the number of componentes an attribute. I understand it can be obtained from `componets_` but it seems to me that it is the sort of value that should be easily available

Reply:
These fields are not private. Also, a `scikit-learn`-compatible estimator should attempt to do the bare minimum in `__init__` (just store the parameters).

Reply:
Okay @vnmabus . I thought the convention was to label public attributes with a trailing underscore. Now I am not sure why some of the fields have that trailing underscore, and others do not. 

If I revert the changes, I will get the `WPS230: Found too many public instance attributes: 11 > 10` warning again
- I can add a noqa comment and ignore it
- I could increase the number of maximum public instance attributes to 11 in the `setup.cfg`

Which one do you think is a better option?

Reply:
The trailing underscore is for public attributes set during `fit` (see https://scikit-learn.org/stable/developers/develop.html).

I think # noqa (with the appropriate error code).

Reply:
Oh, that makes sense.

Great. I will add it in my next push.

Thanks!